src/System.Net.Http/src/System/Net/Http/Headers/HttpHeaders.cs

```diff
@@ -45,7 +45,7 @@ private enum StoreLocation
         }
 
         protected HttpHeaders()
-            : this(HttpHeaderType.All, HttpHeaderType.None)
+            : this(HttpHeaderType.None, HttpHeaderType.All)
```

Comment:
This is probably right based on the discussion in the bug issue.  But it needs some comments around this since it looks confusing to anyone reading this code.

Reply:
There are some test failing because of this change but comments on general directions are appreciated.
I'll add some comments.